src/NITTA/FrontEnds/XMILE/MathParser.hs

```diff
@@ -0,0 +1,82 @@
+{- |
+Module      : NITTA.FrontEnds.XMILE.MathParser
+Description : Parses XMILE math equations
+Copyright   : (c) Aleksandr Penskoi, 2021
+License     : BSD3
+Maintainer  : aleksandr.penskoi@gmail.com
+Stability   : experimental
+-}
+module NITTA.FrontEnds.XMILE.MathParser (
+    parseXmileEquation,
+    calculateDefaultValue,
+    XMExpr (..),
+    XMDuop (..),
+) where
+
+import qualified Data.HashMap.Strict as HM
+import Data.Maybe (fromMaybe)
+import Data.Text as T
+import Text.Parsec
+import Text.Parsec.Expr
+import Text.Parsec.Language
+import Text.Parsec.String
+import Text.Parsec.Token
+import Text.Read hiding (parens)
+
+data XMExpr = Var String | Val Double | Duo XMDuop XMExpr XMExpr
```

Comment:
Maybe you need to name fields for `Duo XMDuop XMExpr XMExpr`?

Reply:
I didn't understand your idea. Can you explain it more precise?